[API design] Add end user's IdP information to host vitals

marko-lisica opened the pull request:
Related to:
- #23236

marko-lisica commented on docs/REST API/rest-api.md:
```diff
@@ -3491,100 +3509,6 @@ Request (`filters` is specified and empty, to delete all hosts):
 
 `Status: 200`
 
-### Get human-device mapping
```
### Dev note:

This endpoint should remain for backwards compatibility. We want to undocumented this as we'll introduce new endpoints that fit better in new IdP to host vitals experience.

marko-lisica commented on docs/REST API/rest-api.md:
```diff
@@ -3491,100 +3509,6 @@ Request (`filters` is specified and empty, to delete all hosts):
-
----
-
-### Update custom human-device mapping
```
### Dev note:

This endpoint should remain for backwards compatibility. We want to undocumented this as we'll introduce new endpoints that fit better in new IdP to host vitals experience.

marko-lisica commented on docs/REST API/rest-api.md:
```diff
@@ -2775,6 +2775,24 @@ Returns the information of the specified host.
     ],
+    "end_user" {
+      "idp_email": "anna@acme.com",
+      "full_name": "Anna Chao",
```
```suggestion
      "full_name": "Anna Chao",
      "id": "f26f8649-1e25-42c5-be71-1b1e6de56d3d",
```

noahtalerman commented on articles/role-based-access.md:
```diff
@@ -96,6 +96,7 @@ GitOps is an API-only and write-only role that can be used on CI/CD pipelines.
 | Edit [macOS setup experience]([https://fleetdm.com/docs/](https://fleetdm.com/guides/macos-setup-experience))\*                                                                                         |          |            | ✅             | ✅    | ✅          |
 | Add and edit identity provider for end user authentication, end user license agreement (EULA), and end user migration workflow\*                                                                                         |          |            |              | ✅    |         |
 | Add and edit Simple Certificate Enrollment Protocol (SCEP) server\*                                                                        |          |            |            | ✅    | ✅      |
+| Connect identity provider (IdP) to add end user informations to hosts\*                                                                        |          |            | ✅          | ✅    |        |
```
```suggestion
| Add end user informations to hosts\*                                                                        |          |            | ✅          | ✅    |        |
```

Do you think this is clear enough? If yes, I think let's move it in the table next to the other hosts rows. Maybe right below "Add/remove manual labels to/from hosts": https://github.com/fleetdm/fleet/pull/26855/files#diff-61a0f8c7cd442fab953eefa601995b52cce1b360efacec358968ef6f2c6a195cR44

noahtalerman commented on handbook/company/pricing-features-table.yml:
```diff
@@ -1116,6 +1116,16 @@
   jamfProtectHasFeature: no
   waysToUse:
     - description: Automatically set admin access to Fleet based on your IDP
+- industryName: Map IdP users to devices (SCIM)
```
I think let's try putting this new row below the existing "Human-endpoint mapping":

![Screenshot 2025-03-11 at 4 24 48 PM](https://github.com/user-attachments/assets/c23db1ac-aaf3-4d57-8e92-c8c747cb3644)

Maybe we call them this?
- "Human-endpoint mapping"
- "Human-endpoint mapping (IdP groups)"

That way, users shopping around will realize they can do mapping for free but they have to pay to get IdP groups mapped.

noahtalerman reviewed (changes requested):
Looking good! Left some feedback re permissions and pricing.

Also, I updated this PR to "Ready for review" to @rachaelshaw knows the API changes are ready for review.

rachaelshaw reviewed (approved):
API lgtm!

marko-lisica commented:
@noahtalerman I changed pricing table and permissions table based on your feedback. I adjusted wording slightly to include SCIM in description, as I heard multiple times folks mentioning that they prefer SCIM over LDAP.

noahtalerman commented on handbook/company/pricing-features-table.yml:
```diff
@@ -326,6 +326,16 @@
     - description: See what servers someone is logged-in on
       moreInfoUrl: https://community.spiceworks.com/topic/138171-is-there-a-way-to-see-what-servers-someone-is-logged-in-on
+- industryName: Human-endpoint mapping (IdP groups)
+  description: Add user information (e.g Full name or group) from identity provider (e.g Okta) to your devices using SCIM protocol.
```
```suggestion
  description: Add user information (e.g full name and groups) from your identity provider (e.g Okta) to your devices using the SCIM protocol.
```

Small capitalization tweak and articles

noahtalerman reviewed (approved):
Nice! LGTM

noahtalerman commented:
FYI @marko-lisica looks like there's some merge conflicts to solve.

marko-lisica commented:
> FYI @marko-lisica looks like there's some merge conflicts to solve.

@noahtalerman @rachaelshaw I just resolved the conflict, could you please approve once more, so I can merge these?

noahtalerman commented:
> @noahtalerman @rachaelshaw I just resolved the conflict, could you please approve once more, so I can merge these?

@marko-lisica looks like we can merge the PR w/o review so I merged it to move quickly.

FYI @rachaelshaw

georgekarrv commented on docs/REST API/rest-api.md:
```diff
@@ -2785,6 +2785,27 @@ Returns the information of the specified host.
+      {
+        "idp_id": "f26f8649-1e25-42c5-be71-1b1e6de56d3d",
+        "idp_email": "anna@acme.com",
+        "idp_full_name": "Anna Chao",
```
@marko-lisica idp_updated_at

marko-lisica replied:
### Dev note

Add a new field to the response: `idp_info_updated_at`


PR to add this to `docs-v4.67.0` branch: ...

marko-lisica replied:
@getvictor We decided to change `idp_email` to `idp_username`.

Here's the PR to update docs: #27764
